### PR TITLE
[FIX] hr_expense: base user access own expenses reports

### PR DESCRIPTION
**Pull request by agr-odoo**
Have a base user with no access to Accounting, Expenses, Employee
Create an expense, input an amount
Save and Create Report

Access Error will raise
You are not allowed to access 'Journal Entry' (account.move) records.

This occurs because of a remaining field in the hr.expense.sheet form
view after refactoring the model

opw-3690493

### CODE REVIEW

**Comment by JulienAlardot** [comment]
Thx a lot for this fix 
Could you add a test?

**Comment by agr-odoo** [comment]
@JulienAlardot 
Test added, not sure if there is a more straightforward way to check access rights so any suggestion is appreciated

File: addons/hr_expense/tests/test_expenses_access_rights.py [hunk 1]
```diff
@@ -116,3 +119,33 @@ def test_expense_sheet_access_rights_refuse(self):
...
+        })
+
+        expense_sheet = self.env['hr.expense.sheet'].with_user(user).create({
+            'name': 'First Expense for employee',
+            'employee_id': expense_employee.id,
+            'journal_id': self.company_data['default_journal_purchase'].id,
+            'accounting_date': '2017-01-01',
+            'expense_line_ids': [
+                (0, 0, {
+                    # Expense without foreign currency but analytic account.
```
Comment (JulienAlardot) [inline addons/hr_expense/tests/test_expenses_access_rights.py:141]:
```suggestion
            'expense_line_ids': [
                Command.create({
                    # Expense without foreign currency but analytic account.
```
Please use helpers

File: addons/hr_expense/tests/test_expenses_access_rights.py [hunk 2]
```diff
@@ -116,3 +119,33 @@ def test_expense_sheet_access_rights_refuse(self):
         expense_sheet.with_user(self.expense_user_manager).action_refuse_expense_sheets()
         expense_sheet.with_user(self.expense_user_manager)._do_refuse('failed')
         self.assertRecordValues(expense_sheet, [{'state': 'cancel'}])
+
+    def test_expense_sheet_access_rights_user(self):
+        # The expense user (no rights) is able to create and read sheet
+
```
Comment (JulienAlardot) [inline addons/hr_expense/tests/test_expenses_access_rights.py:125]:
```suggestion
    def test_expense_sheet_access_rights_user(self):
        # The expense base user (without other rights) is able to create and read sheet

```

File: addons/hr_expense/tests/test_expenses_access_rights.py [hunk 3]
```diff
@@ -1,12 +1,15 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import odoo.tests
+
 from odoo.addons.hr_expense.tests.common import TestExpenseCommon
 from odoo.exceptions import AccessError, UserError
-from odoo.tests import tagged
+from odoo.tests import tagged, new_test_user
```
Comment (JulienAlardot) [inline addons/hr_expense/tests/test_expenses_access_rights.py:8]: Please import HttpCase from odoo.test instead

**Review by JulienAlardot — CHANGES_REQUESTED** [review]

**Comment by agr-odoo** [comment]
@JulienAlardot PR updated with suggestions, thanks!

**Comment by JulienAlardot** [comment]
@epictete LGTM

**Review by JulienAlardot — APPROVED** [review]

File: addons/hr_expense/static/tests/tours/expense_upload_tours.js [hunk 4]
```diff
@@ -76,3 +76,29 @@
...
+        {
+            content: "Go to Expense",
+            trigger: '.o_app[data-menu-xmlid="hr_expense.menu_hr_expense_root"]',
+        },
+        {
+            content: "Go to My Expenses",
+            trigger: 'button[data-menu-xmlid="hr_expense.menu_hr_expense_my_expenses"]',
+        },
+        {
+            content: "Go to My Expenses to Report",
```
Comment (epictete) [inline addons/hr_expense/static/tests/tours/expense_upload_tours.js:93]:
```suggestion
            content: "Go to My Reports",
```

File: addons/hr_expense/static/tests/tours/expense_upload_tours.js [hunk 5]
```diff
@@ -76,3 +76,29 @@
...
+            trigger: 'a[data-menu-xmlid="hr_expense.menu_hr_expense_sheet_my_reports"]',
+        },
+        {
+            content: "Go to My Expenses to Report",
```
Comment (epictete) [inline addons/hr_expense/static/tests/tours/expense_upload_tours.js:97]:
```suggestion
            content: "Go to First Expense for employee",
```

File: addons/hr_expense/static/tests/tours/expense_upload_tours.js [hunk 6]
```diff
@@ -76,3 +76,29 @@
...
+            trigger: 'td[data-tooltip="First Expense for employee"]',
+        },
+        {
+            content: "Check Submit to Manager Button",
```
Comment (epictete) [inline addons/hr_expense/static/tests/tours/expense_upload_tours.js:101]:
```suggestion
            content: "Click Submit to Manager Button",
```

**Review by epictete — COMMENTED** [review]
Tiny cosmetic suggestions :wink:

File: addons/hr_expense/static/tests/tours/expense_upload_tours.js (hunk 4 above)
Comment (agr-odoo) [reply addons/hr_expense/static/tests/tours/expense_upload_tours.js]: Thanks, I always forget to double checks comments

**Review by epictete — APPROVED** [review]
Thanks for the update :ok: 

@robodoo r+

**Pull request closed without merging**